Sources/NIOWebSocket/WebSocketUpgrader.swift

```diff
@@ -0,0 +1,126 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//
+//  WebSocketUpgrader.swift
+//  NIOWebSocket
+//
+//  Created by Cory Benfield on 14/02/2018.
+//
+
+import class Foundation.NSNull // for string operations
```

Comment:
do we really need this?

Reply:
Not anymore, with `trimWhitespace` being included in `getCanonicalForm`.

Reply:
@Lukasa if you feel like it you could add the websockets stuff to the test that tests that we don't link Foundation